frontend/features/productList/components/sections/FilterableProductsSection/SearchInput.tsx

```diff
@@ -38,7 +53,7 @@ export const SearchInput = chakra(({ className }: SearchInputProps) => {
                bg="white"
                placeholder="Search"
                tabIndex={0}
-               onChange={(event) => search(event.currentTarget.value)}
+               onChange={onSearchChange}
```

Comment:
Why the added complexity instead of just using a `maxLength` prop?

Reply:
That's right, much more elegant solution! I've also removed the callback memoization as probably isn't doing that much